### PR TITLE
AEROGEAR-8774 post client data to go mobile security service

### PR DESCRIPTION
**Pull request by austincunningham**
### Description
Merging to feature branch `app-security`
# What
PR Post modified device data and app data to the security services app-service module

# How
returns a promise from axios

# Why
Needed for the Mobile security service Go server app-security integration
https://issues.jboss.org/browse/AEROGEAR-8774

# Verification
## Prerequisites 
- node 10
- npm 6
- install ionic https://ionicframework.com/docs/installation/cli
- install android sdk 26 http://www.androiddocs.com/sdk/installing/adding-packages.html

## Steps
- clone this repo
- checkout branch AEROGEAR-8774
- install the aerogear-js-sdk
```bash
npm i && npm run bootstrap && npm run build
```

- clone https://github.com/austincunningham/ionic-showcase.git
- checkout branch AEROGEAR-8773
- change the package.json to point at your local clone of aerogear-js-sdk e.g.

image

> Note you can try and use npm link but I didn't have much joy with it 
- install ionic-showcase
```
npm i
```
- build the android apk
```
ionic cordova build android
```
- copy the apk to your android device emulation
- open the app on the emulator
- follow these steps to attach your chrome developer console to your emulation https://developers.google.com/web/tools/chrome-devtools/remote-debugging/

- start the mobile security services db as outline here https://github.com/aerogear/mobile-security-service#131-by-using-the-docker-image-of-this-project
- start the go mobile security service as outline here https://github.com/aerogear/mobile-security-service#14-starting-the-server

- add the app to the app table
set the `id` field with a uuid you can generate one here https://www.uuidgenerator.net/
set the `app_id` with `org.aerogear.ionic.showcase`
set the `app_name` to any string

- In the console you should see the INIT Data and the metrics payload to verify this

image

##### Checklist

- [x] `npm test` passes
- [x] `npm run build` works

### CODE REVIEW

File: packages/security/src/appSecurity/AppSecurity.ts [hunk 1]
```diff
@@ -33,6 +32,20 @@ export class AppSecurity {
...
+    const clientInitResponse = this.getClientData()
+    .then(metricsPayload => {
+      const initPayload: {[key: string]: any; } = {};
+      // build the initPayload
+      initPayload.deviceId = metricsPayload.clientID;
+      initPayload.appId = metricsPayload.app.appId,
+      initPayload.deviceType = metricsPayload.device.platform;
+      initPayload.deviceVersion = metricsPayload.device.platformVersion;
+      initPayload.version = metricsPayload.app.appVersion;
+      return axios.post(this.internalConfig.url, initPayload);
```
Comment (wtrocki) [inline packages/security/src/appSecurity/AppSecurity.ts]: [Question] This is probably service we will require better flexibility that metrics publisher do not offer for us?

[Question] Does URL contain the full path to service?

Comment (austincunningham) [reply packages/security/src/appSecurity/AppSecurity.ts]: [Answer] I think so
[Answer] yes see my version of showcase config file at https://github.com/austincunningham/ionic-showcase/blob/383dc01008d62b2e8af92a0888dc481c381b1788/src/mobile-services.js#L10

Comment (wtrocki) [reply packages/security/src/appSecurity/AppSecurity.ts]: @austincunningham Yes. I got into impression that there will be more than one endpoint but I think your case is simple as there will be the only endpoint called so the entire path can be used etc. Great work overall.

**Comment by wtrocki** [comment]
@austincunningham `npm install axios` needed to merge. It sents metrics as expected. Good work!

File: packages/security/src/appSecurity/AppSecurity.ts (hunk 1 above)
Comment (austincunningham) [reply packages/security/src/appSecurity/AppSecurity.ts]: @wtrocki I taught there was more that one as well, but I checked the acceptance criteria and just needed this one.

**Pull request merged**